java-extension/com.microsoft.java.test.plugin/src/main/java/com/microsoft/java/test/plugin/util/TestGenerationUtils.java

```diff
@@ -152,16 +163,280 @@ public static WorkspaceEdit generateTests(List<Object> arguments, IProgressMonit
+            buf.append(constructMethodStub(testUnit, testKind, method, lineDelimiter)).append(lineDelimiter);
+        }
+        buf.append("}").append(lineDelimiter);
+        return buf.toString();
+    }
+
+    private static String constructMethodStub(ICompilationUnit testUnit, TestKind testKind,
+            String method, String lineDelimiter) {
+        final StringBuilder buf = new StringBuilder();
+        buf.append("@Test").append(lineDelimiter);
+        if (testKind == TestKind.JUnit) {
+            buf.append("public ");
+        }
+        final String methodName = getTestMethodName(method);
+        buf.append("void ").append(methodName).append("() {").append(lineDelimiter).append(lineDelimiter).append("}");
+        final String methodContent = buf.toString();
+        // TODO: get test unit options directly
+        return CodeFormatterUtil.format(CodeFormatter.K_STATEMENTS, methodContent, 1,
+                lineDelimiter, testUnit.getJavaProject().getOptions(true));
+    }
+
+    private static IClasspathEntry getTestClasspathEntry(ICompilationUnit unit) throws JavaModelException {
+        final IJavaProject javaProject = unit.getJavaProject();
+        // In most cases, this is the classpath entry used for testing, we first find the target entry by hard-code
+        // to avoid go into the generated entries.
+        final IClasspathEntry testEntry = javaProject.getClasspathEntryFor(
+            javaProject.getPath().append("src/test/java"));
+        if (ProjectTestUtils.isTestEntry(testEntry)) {
+            return testEntry;
+        }
+
+        final IClasspathEntry[] entries = javaProject.readRawClasspath();
+        for (final IClasspathEntry entry : entries) {
+            if (ProjectTestUtils.isTestEntry(entry)) {
+                return entry;
+            }
+        }
+
+        return entries[0];
+    }
+
+    private static String getTestFullyQualifiedName(ITypeBinding typeBinding, IJavaProject project,
+            IClasspathEntry testEntry) throws JavaModelException {
+        String promptName = typeBinding.getBinaryName() + "Tests";
+        final ICompilationUnit testCompilationUnit = getTestCompilationUnit(project, testEntry, promptName);
+        if (!testCompilationUnit.exists()) {
+            promptName = typeBinding.getBinaryName() + "Test";
+        }
+
+        final String fullyQualifiedName = (String) JUnitPlugin.askClientForInput(
+            "Please type the target test class name", promptName);
+        if (fullyQualifiedName == null) {
+            return null;
+        }
+
+        if (fullyQualifiedName.charAt(fullyQualifiedName.length() - 1) == '.') {
+            JavaLanguageServerPlugin.getInstance().getClientConnection().showNotificationMessage(MessageType.Error,
+                    "Invalid Java class name: " + fullyQualifiedName);
+            return null;
+        }
+
+        final String[] identifiers = fullyQualifiedName.split("\\.");
```

Comment:
validating the qualified name should be a problem that has already been resolved.

How about taking a look at `org.eclipse.jdt.core.JavaConventions`?

Reply:
That's more convinient!